### DISCUSSION TITLE
How to get ip-adapter_face_id_plus ControlNet preprocessor?

### DISCUSSION THREAD

**Discussion by ansorre** [lllyasviel/stable-diffusion-webui-forge#524] [Q&A]
Can't find a way to get ControlNet preprocessor:
ip-adapter_face_id_plus 
And, sorry, no, InsightFace+CLIP-H produces way different images compared to what I get on a1111 with ip-adapter_face_id_plus even using the same model.

**Comment by leomaxwell973** [comment]
This post may shed some light

https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/178#discussioncomment-8572388

perhaps you used one version or the other? its actually bit shocking how the preprocessor has gone back and forth 3-ways, never noticed, though then again, with the size of BigG and OOM issues when multi-...anything in SDXL mode makes sense but still ping-pong.



anyway afaik your target should be Clip-VitH (...or if you really want it BigG)
if it's not sending you a download location heres ones I found:

https://huggingface.co/laion/CLIP-ViT-H-14-laion2B-s32B-b79K

https://huggingface.co/comfyanonymous/clip_vision_g/tree/main
(had issues finding the correct G, in safetensors format, renaming this one should be fine, i think, see below.)



as for the insightful face stuff.. not entirely sure how that works? not without reading more later that is, think you just need an extension installed is my best guess. though by the sounds of it, you were probably accustomed to just a pure G/H config (hopefully H) so probably no need to worry in that case.

food for thought, paths and the files names forge uses:
models\ControlNetPreprocessor\CLIP-ViT-H-14.safetensors & CLIP-ViT-bigG.safetensors

In case you need to change them from clip_h.pth/safetensors or clip_g.pth/safetensors as is often the case -_-

edit: on that note, may be worth a peek if you already have a copy named the wrong thing in the wrong location, for control-nets purposes at least, could just copy/sym-link it to the path it wants with the name it wants then should be gtg.

**Reply by ansorre** [reply]
Thanks @leomaxwell973 but I have problems with the table you point:

[https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/178#discussioncomment-8572388](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/178#discussioncomment-8572388)

Is "IPadapter" col referring to the models and "Img encoder" col to the preprocessor? Because there is no preprocessor in my forge installation called just "ViT-H" as reported in that table for "FaceID Plus v2" model for SDXL. So I for example tried "CLIP-ViT-H (IPAdapter)" or even "CLIP-ViT-bigG (IPAdapter)" but I get this error:
 Exception: InsightFace must be provided for FaceID models.
 But, as I stated in the original message, using "InsightFace+CLIP-H (IPAdapter)" does not result in the same images I get on a1111 with "ip-adapter_face_id_plus" even using the same model (ip-adapter-faceid-plusv2_sdxl).

**Comment by SlowMan69** [comment]
> Can't find a way to get ControlNet preprocessor: ip-adapter_face_id_plus And, sorry, no, InsightFace+CLIP-H produces way different images compared to what I get on a1111 with ip-adapter_face_id_plus even using the same model.

Were you able to solve this issue? I have the same issue, I am using Forge.
